### ISSUE TITLE
Vaadin 13: Initial property-change with new API

### ISSUE THREAD

**Issue by Springrbua** [vaadin/flow#5152]
**Description of the bug:**
In Vaadin 13, the method `synchronizeProperty` in `Element` has been deprecated.
The new API is `addPropertyChangeListener(String, String, PropertyChangeListener)`.
However, with this new API, the initial `PropertyChangeEvent` is not fired.
I am using `Component`s, which rely on the `iron-resize` event to get the `clientWidth`/`clientHeight` and update their behavior/appearence accordingly.
Using the new API, I don't get their initial `clientWidth`/`clientHeight`.

**Steps to reproduce:**

1.  Download the [Project Base for Vaadin Flow (pre-release)](https://vaadin.com/start/pre-release/project-base)
2. In the `MainView.java` add the following code:
```

public class MainView extends VerticalLayout {
    private SplitLayout layout;
    private Span primary;
    public MainView() {
        layout = new SplitLayout();
        layout.setSizeFull();
        layout.addToPrimary(primary = new Span("Total width: "));
        layout.addToSecondary(new Span("Secondary"));
        add(layout);
    }
    protected void onAttach(AttachEvent attachEvent) {
        layout.getElement().addPropertyChangeListener("clientWidth", "iron-resize", event -> primary.setText("Total width: " + event.getValue()));
    }
}
```
Basically I add a `SplitLayout`, which has the `IronResizableBehavior`. Then I add a `PropertyChangeListener` for the `clientWidth`, which should be triggered by the `iron-resize`.
The listener simply sets the `clientWidth` of the `SplitLayout` as text to the primary content.

If you run this project, you will notice, that the text is only set, if you resize the window, or drag the splitter, but the initial value is missing.

**Expected behaviour**
If you use the deprecated API instead, you will get the initial event.
To test this, simply replace the code in `onAttach` with this code:
```
layout.getElement().synchronizeProperty("clientWidth", "iron-resize");
layout.getElement().addPropertyChangeListener("clientWidth", event -> primary.setText("Total width: " + event.getValue()));
```

**Comment by denis-anisimov** [comment]
Looks like a bug in the newly implemented functionality `Element::addPropertyChangeListener`.
In this scenario it should work in the same way as synchronized property with a property listener on event.

**Comment by Legioth** [comment]
The reason for this bug is that by accident, `synchronizeProperty` causes the DOM listener to be added to the element before it is attached to a parent, whereas `addEvenListener` is added to the element only after the element is attached.

In most cases, this has no impact but in this particular case there is a difference because `iron-resize` is fired through `connectedCallback`.

We should most likely fix the ordering of adding event listeners so that they always happen before the element is attached.
